src/test/java/com/synopsys/integration/stepworkflow/AbstractConsumingSubStepTest.java

```diff
@@ -0,0 +1,87 @@
+package com.synopsys.integration.stepworkflow;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+public class AbstractConsumingSubStepTest {
+
+    private final String verificationData = "Verification output";
+    private final RuntimeException verificationException = new RuntimeException("Verification runtime exception");
+    private final SubStepResponse<Object> subStepResponse_successOutput = new SubStepResponse<>(true, verificationData, verificationException);
+
+    private static Stream<Arguments> populateSuccessTests() {
+        final String successData = "Success Data";
+        RuntimeException runtimeException = new RuntimeException("Success runtime exception");
+        return Stream.of(
```

Comment:
These arguments appear to be for constructing SubStepResponses-- why not just provide a stream of SubStepResponses to test against?

Reply:
Purely to help anyone debugging failed tests. When data points are passed, the name of the test within the output html will be something like:

```
[1] true, , null
[2] true, , java.lang.RuntimeException: Success runtime exception
```
When passing in an object, the name of the test will be something like:

```
[1] com.synopsys.integration.stepworkflow.SubStepResponse@59816976
[2] com.synopsys.integration.stepworkflow.SubStepResponse@3be33474
```

If the benefit of displaying the data points in the html is not greater than what it detracts from the code, I will update this asap as well.

Reply:
Data in HTML sounds compelling to me.